## Getting the Data

In [1]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np

from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score
from sklearn.metrics import roc_auc_score

In [2]:
SourceFile = 'https://raw.githubusercontent.com/jbrownlee/Datasets/master/haberman.csv'
df = pd.read_csv(SourceFile, names=['Age','Year of Operation','Number of PAN','Survival Status'])
df.head()

Age  Year of Operation  Number of PAN  Survival Status
0   30                 64              1                1
1   30                 62              3                1
2   30                 65              0                1
3   31                 59              2                1
4   31                 65              4                1

## No.1

Bagi kedua data ini menjadi data training dan data test dengan test_size=0.25.

In [3]:
# Memisahkan dataset menjadi predictor dan target
X = df.drop("Survival Status", axis=1)
y = df['Survival Status']

print("X:", X.shape)
print("y:", y.shape)

X: (306, 3)
y: (306,)


In [4]:
#memisahkan dataset train dengan test
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size = 0.25, random_state=27)

print("X_train:", X_train.shape)
print("X_test:", X_test.shape)
print("y_train:", y_train.shape)
print("y_test:", y_test.shape)

X_train: (229, 3)
X_test: (77, 3)
y_train: (229,)
y_test: (77,)


In [5]:
from sklearn.preprocessing import StandardScaler, MinMaxScaler

scaler= StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

## No.2

Pelajari tentang metrics roc_auc_score kemudian buatlah model dan evaluasi dengan menggunakan teknik cross-validation dengan scoring 'roc_auc'.

In [6]:
# model
decision_tree = DecisionTreeClassifier() 
model_dt = decision_tree.fit(X_train, y_train)

#evaluasi
r_a_score = cross_val_score(decision_tree, X, y, scoring='roc_auc', cv=10)

## No.3

Berapa score rata-rata dari model dengan teknik cross-validation tersebut?

In [7]:
#score
print("Score Mean:", round(r_a_score.mean(),2))

Score Mean: 0.49


## No.4

Prediksi data test dengan model yang telah kalian buat.

In [8]:
from sklearn.metrics import accuracy_score
y_test_pred = model_dt.predict(X_test)
test = accuracy_score(y_test, y_test_pred)
print("acc_score:", round(test, 2))

acc_score: 0.7


## No.5

Bagaimana hasil confusion matrix dari hasil prediksi tersebut?

In [9]:
from sklearn.model_selection import cross_val_predict
from sklearn.metrics import confusion_matrix

confusion_matrix(y_test, y_test_pred)

array([[44, 13],
       [10, 10]])

## No.6

Bagaimana classification report dari hasil prediksi tersebut?

In [10]:
from sklearn.metrics import precision_score, recall_score, f1_score

print("Precision:", round(precision_score(y_test, y_test_pred),2))
print("Recall:", round(recall_score(y_test, y_test_pred),2))
print("F1_score:", round(f1_score(y_test, y_test_pred),2))

Precision: 0.81
Recall: 0.77
F1_score: 0.79


## No.7

Seberapa baik model anda dalam memprediksi seorang pasien mempunyai status positive?

model memiliki akurasi sebesar 82% dalam memprediksi seorang pasien mempunyai status positive

## No.8

Seberapa baik model anda dalam memprediksi seorang pasien mempunyai status negative?

model memiliki akurasi sebesar 79% dalam memprediksi seorang pasien mempunyai status negative